frontend/src/Context/AudioOutputProvider/useAudioOutput/useAudioOutput.tsx

```diff
@@ -15,26 +18,41 @@ export type AudioOutputContextType = {
  * @returns {AudioOutputContextType} audioOutput context
  */
 const useAudioOutput = (): AudioOutputContextType => {
-  const [audioOutput, setAudioOutput] = useState<AudioDeviceId>();
+  const [currentAudioOutputDevice, setCurrentAudioOutputDevice] = useState<AudioDeviceId>();
+  const { mediaDevices } = window.navigator;
 
-  useEffect(() => {
-    getActiveAudioOutputDevice().then((audioOutputDevice) => {
+  const updateCurrentAudioOutputDevice = useCallback(() => {
+    getVonageActiveAudioOutputDevice().then((audioOutputDevice) => {
       if (audioOutputDevice.deviceId) {
-        setAudioOutput(audioOutputDevice.deviceId);
+        setCurrentAudioOutputDevice(audioOutputDevice.deviceId);
       }
     });
   }, []);
 
-  const updateAudioOutput = useCallback(async (deviceId: AudioDeviceId) => {
+  useEffect(() => {
+    updateCurrentAudioOutputDevice();
+  }, [updateCurrentAudioOutputDevice]);
+
+  useEffect(() => {
+    // Add an event listener to update device list when the list changes (such as plugging or unplugging a device)
+    mediaDevices.addEventListener('devicechange', updateCurrentAudioOutputDevice);
+
+    return () => {
+      // Remove the event listener when component unmounts
+      mediaDevices.removeEventListener('devicechange', updateCurrentAudioOutputDevice);
+    };
+  }, [mediaDevices, updateCurrentAudioOutputDevice]);
```

Comment:
I added this event listener to update the current device when the device list changes. This keeps the UI in sync if the current device was removed